parlai/crowdsourcing/tasks/wizard_of_internet/acceptability.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import numpy as np
+from typing import Iterable, List
+from nltk.stem import PorterStemmer
+from parlai.crowdsourcing.utils.acceptability import (
+    AcceptabilityChecker,
+    normalize_answer,
+)
+import parlai.utils.logging as logging
+
+
+# Bad persona violations
+PERSONA_REPEATS_PROMPT = 'repeated the prompt text'
+ASKED_WIZARD_QUESTION = 'asked wizard in the persona details'
+COPIED_EXTENDED_PERSONA = 'extended persona copies the main persona'
+GENERIC_EXTENDED_PERSONA = 'extended persona is generic'
+
+QUESTION_PHRASE = 'what is your'
+
+# Wizard knowledge violations
+DEFAULT_KNOWLEDGE_OVERLAP_THRESHOLD = 0.05
+
+POOR_SEARCH_QUERIES = 'poor search queries'
+IRRELEVANT_SEARCH__QUERIES = 'irrelevant search terms'
+NOT_ENOUGH_SEARCH = 'not enough selected knowledge sources'
+SELECTED_SHORT_PIECES = 'short knowledge pieces selected.'
+LOW_KNOWLEDGE_OVERLAP = 'low knowledge overlap'
+
+
+def tokenize_text(text, stemmer, as_set=True):
+    text = normalize_answer(text)
+    tokens = [stemmer.stem(word) for word in text.split(' ')]
+    if as_set:
+        tokens = set(tokens)
+    return tokens
+
+
+def overlap_ratios(a: set, b: set) -> float:
+    """
+    Calculates the Jacard distance between two sets.
+    """
+    overlap = a.intersection(b)
+    union = a.union(b)
+    return len(overlap) / (len(union) + 0.001)
+
+
+def is_valid_agent_chat_message(message, agent_id):
+    return (
+        message.get('text')
+        and message.get('id') == agent_id
+        and not message.get('is_search_query', False)
+    )
+
+
+def bad_persona(persona, stemmer):
+    """
+    Check for poor persona selection by apprentice.
+    """
+    persona_parts = persona.split('\n')
+
+    # It is not from the persona selection ones (personas used during the pilot).
```

Comment:
do we need to include this? i.e. are we supporting personas from the pilot?

Reply:
Actually this one is because of the extra explanation that apprentice adds, or the locations that we are adding. Each of those are going be a separate line.